src/internal/partial_solution.rs

```diff
@@ -274,6 +263,9 @@ impl<P: Package, V: Version> PartialSolution<P, V> {
                 break;
             }
         }
+        // TODO: there is a bug here? What happens if there no previous dated_assignment for
+        // package? satisfier_map should be left unchanged and the max index is the one of the
+        // satisfier, which should be outside the range of the previous_assignments slice?
```

Comment:
`previous_assignments` slice is now made with `0..=satisfier_idx` so that this call can work, for exactly this reason.

Reply:
This is a bit outside the context of this PR, but since we're discussing this ...
If there is no previous dated_assignment for package, the max index for package is kept unchanged as the one of the satisfier. As a result the satisfier package will be the one selected by
```rust
previous_assignments[*satisfier_map.values().max().unwrap()]
```
And thus the decision level returned will be the one of the satisfier. But as stated in pub guide,

> Let previousSatisfierLevel be previousSatisfier's decision level, or decision level 1 if there is no previousSatisfier.

So in that precise case, it should return 1. So maybe we should use instead `0..satisfier_idx` for previous assignments and in the end, return
```rust
        previous_assignments
            .get(*satisfier_map.values().max().unwrap())
            .map(|a| a.decision_level)
            .unwrap_or(DecisionLevel(1))
            .max(DecisionLevel(1))
```